tests/BasicSearchTests.FunctionalTests.Core/V2SearchFunctionalTests.cs

```diff
@@ -37,5 +38,60 @@ public class V2SearchFunctionalTests : BaseFunctionalTests
             Assert.True(result.TotalHits.HasValue && result.TotalHits.Value > 0, "No results found, should find atleast some results for empty string query.");
             Assert.NotNull(result.Data);
         }
+
+        [Fact]
+        public void ShouldThrowExceptionForInvalidSearchResults()
```

Comment:
That is correct. The real functional tests are `EmptyQueryResultsHaveRequiredProperties` and `ExistingPackageResultsHaveRequiredProperties`. Both of these functional tests assume that invalid results, should they happen, will cause a `JsonSerializationException` exception to be thrown. This `ShouldThrowExceptionForInvalidSearchResults` test is simply a sanity check that validates that assumption.